src/bptl/work_units/kownsl/tasks.py

```diff
@@ -335,3 +295,40 @@ def get_email_details(task: BaseTask) -> dict:
         "template": template,
         "senderUsername": [requester],
     }
+
+
+@register
+def set_review_request_metadata(task: BaseTask) -> dict:
+    """
+    Set the metadata for a Kownsl review request.
+
+    Metadata is a set of arbitrary key-value labels, allowing you to attach extra data
+    required for your process routing/handling.
+
+    **Required process variables**
+
+    * ``kownslReviewRequestId``: the identifier of the Kownsl review request.
+    * ``metadata``: a JSON structure holding key-values of the metadata. This will be
+      set directly on the matching review request. Example:
+
+      .. code-block:: json
+
+            {
+                "processInstanceId": "aProcessInstanceId"
+            }
+
+    **Sets no process variables**
+
+    """
+    variables = task.get_variables()
+    review_request_id = check_variable(variables, "kownslReviewRequestId")
+    metadata = check_variable(variables, "metadata")
+
+    client = get_client(task)
+    client.partial_update(
+        "reviewrequest",
+        data={"metadata": metadata},
+        uuid=review_request_id,
+    )
+
+    return {}
```

Comment:
Why does this return an empty dict?

Reply:
Because a task must always return a dict, that's the contract of the task. The return value of the task is set as process variables, but this task does not set any process variables at all.